components/tidb_query_expr/src/impl_time.rs

```diff
@@ -150,6 +150,26 @@ pub fn to_seconds(ctx: &mut EvalContext, t: &DateTime) -> Result<Option<Int>> {
     Ok(Some(t.second_number()))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn date_diff(
+    ctx: &mut EvalContext,
+    from_time: &DateTime,
+    to_time: &DateTime,
+) -> Result<Option<Int>> {
+    if from_time.invalid_zero() {
```

Comment:
No need to check, it's already checked in codec.

Reply:
In general, we need to handle errors when we construct DateTime or Duration. And we can assume the arguments are all vaild.

Reply:
done

Got it. While for `to_seconds` and `month_name` the `invalid_zero` check also exist, and there are invalid `DateTime` test cases for these two methods. Is it necessary?